// @mui/material-next/node/styles/defaultTheme.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getThemeWithVars = exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
var _utils = require("@mui/utils");
var _extendTheme2 = _interopRequireDefault(require("./extendTheme"));
const _excluded = ["colorSchemes", "opacity", "overlays", "shape", "md3", "ref", "sys", "palette"],
  _excluded2 = ["mode", "colorScheme", "ref", "sys"];
const getThemeWithVars = themeInput => {
  const _extendTheme = (0, _extendTheme2.default)(themeInput),
    {
      colorSchemes,
      opacity,
      overlays,
      shape,
      md3,
      ref,
      sys,
      palette: paletteInput
    } = _extendTheme,
    restTheme = (0, _objectWithoutPropertiesLoose2.default)(_extendTheme, _excluded);
  const colorSchemePalette = (0, _utils.deepmerge)(colorSchemes[(paletteInput == null ? void 0 : paletteInput.colorScheme) || 'light'].palette, paletteInput);
  const {
      mode = 'light',
      colorScheme = 'light',
      // @ts-ignore md3 specific token
      ref: colorSchemeRef,
      // @ts-ignore md3 specific token
      sys: colorSchemeSys
    } = colorSchemePalette,
    palette = (0, _objectWithoutPropertiesLoose2.default)(colorSchemePalette, _excluded2);
  return (0, _extends2.default)({
    opacity,
    overlays,
    shape,
    md3,
    ref: (0, _extends2.default)({}, ref, colorSchemeRef),
    sys: (0, _extends2.default)({}, sys, colorSchemeSys)
  }, restTheme, {
    colorSchemes: (0, _extends2.default)({}, colorSchemes, {
      [colorScheme]: {
        palette,
        ref: colorSchemeRef,
        sys: colorSchemeSys
      }
    }),
    palette: (0, _extends2.default)({}, palette, {
      mode,
      colorScheme
    }),
    vars: {
      opacity,
      overlays,
      shape,
      ref: (0, _extends2.default)({}, ref, colorSchemeRef),
      sys: (0, _extends2.default)({}, sys, colorSchemeSys),
      md3,
      palette
    }
  });
};
exports.getThemeWithVars = getThemeWithVars;
const defaultTheme = getThemeWithVars();
var _default = defaultTheme;
exports.default = _default;
